### PR TITLE
Java: Reuse arguments rule at methodCall rule (#3430)

### LINKED ISSUE #3430: [Java] `methodCall` rule should contains `arguments` rule
At the  moment, rule `method` at [Java parser](https://github.com/antlr/grammars-v4/blob/master/java/java/JavaParser.g4) looks like:
```antlr
methodCall
    : identifier '(' expressionList? ')'
    | THIS '(' expressionList? ')'
    | SUPER '(' expressionList? ')'
    ;
```
But here in grammar there is an `arguments` rule
```antlr
arguments
    : '(' expressionList? ')'
    ;
```
Which is used in many other rules, e.g. `classCreatorRest`:
```antlr
classCreatorRest
    : arguments classBody?
    ;
```

---
It looks very illogical not to reuse existing rules.
So, to discard this logical misunderstanding, the `methodCall` rule should contains `arguments` rule.
As following:
```antlr
methodCall
    : identifier arguments
    | THIS arguments
    | SUPER arguments
    ;
```

### PR DESCRIPTION
**Pull request by neNerower**
Fix #3430

### CODE REVIEW

**Comment by KvanTTT** [comment]
There are some errors on remote run. Please fix them.

File: java/java/JavaParser.g4 [hunk 1]
```diff
@@ -582,9 +582,9 @@ expressionList
     ;
 
 methodCall
-    : identifier '(' expressionList? ')'
-    | THIS '(' expressionList? ')'
-    | SUPER '(' expressionList? ')'
+    : identifier arguments
+    | THIS arguments
+    | SUPER arguments
```
Comment (KvanTTT) [inline java/java/JavaParser.g4:587]: Does it make sense to simplify it in the following way?

```suggestion
    : (identifier | THIS | SUPER) arguments
```

Comment (neNerower) [reply java/java/JavaParser.g4:587]: Yes, it can make sense

**Comment by neNerower** [comment]
@KvanTTT 
All tests except of Go (as was mentioned at [this comment](https://github.com/antlr/grammars-v4/pull/3456#issuecomment-1557087030)) were successful.

**Comment by kaby76** [comment]
> @KvanTTT All tests except of Go (as was mentioned at [this comment](https://github.com/antlr/grammars-v4/pull/3456#issuecomment-1557087030)) were successful.

The Go target driver needs to change with the Antlr 4.13.0 release. Please merge https://github.com/antlr/grammars-v4/pull/3469. It contains the fix for Go, and fixes for several other problems, which we missed with https://github.com/antlr/grammars-v4/pull/3467

**Comment by neNerower** [comment]
> > @KvanTTT All tests except of Go (as was mentioned at [this comment](https://github.com/antlr/grammars-v4/pull/3456#issuecomment-1557087030)) were successful.
> 
> The Go target driver needs to change with the Antlr 4.13.0 release. Please merge #3469. It contains the fix for Go, and fixes for several other problems, which we missed with #3467

Done

**Comment by teverett** [comment]
@neNerower please resolve the conflicts

**Review by KvanTTT — APPROVED** [review]

**Pull request merged**